dart/lib/src/error_cause_extractor.dart

```diff
@@ -0,0 +1,21 @@
+class CauseExtractor<T> {
+  Object? cause(T error) { return null; }
+}
+
+class ErrorCauseExtractor {
+  ErrorCauseExtractor(this.causeExtractorsByType);
+
+  final Map<Type, CauseExtractor> causeExtractorsByType;
+
+  List<Object> flatten(errorWithCause) {
+    var allErrors = <Object>[];
+    var current = errorWithCause;
+
+    while (current != null) {
+      allErrors.add(current);
```

Comment:
Something to pay attention here is the sorting of the list since the SDK should report in the correct order.
Check the Java implementation https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/SentryExceptionFactory.java#L113
I dont recall now if the root cause is the first or the last in the list, can double check if needed.

Reply:
Root cause on Java is the [first in the list](https://github.com/getsentry/sentry-java/blob/68dfbfd9b01772f8cf9013dae7cbf14c3f43c378/sentry/src/main/java/io/sentry/SentryExceptionFactory.java#L126) and then we go down deeper through the cause properties. We do it the same way and the order of extraction is covered by the tests in `exception_cause_extractor_test.dart`